tools/approval-service/internal/service/ttleventcache_test.go

```diff
@@ -0,0 +1,140 @@
+package service_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gravitational/shared-workflows/tools/approval-service/internal/service"
+)
+
+var (
+	eventIDForTest           = "event-id-for-test"
+	eventIDInProgressForTest = "event-id-in-progress-for-test"
+)
+
+// TestTryAddConcurrency ensures that under a burst of concurrent TryAdds for the same key
+// exactly one caller wins (returns true) and others are rejected within the TTL
+func TestTryAddConcurrency(t *testing.T) {
+	// Arrange
+	ttl := 200 * time.Millisecond
+	ec, cleanupFunc, err := service.MakeTTLEventCache(ttl)
+	if err != nil {
+		t.Fatalf("MakeTTLCache error: %v", err)
+	}
+	defer func() {
+		_ = cleanupFunc
+	}()
+
+	const goroutines = 500
+	var wg sync.WaitGroup
+
+	var wins int32
+	var mu sync.Mutex
+
+	// Act
+	for i := 0; i < goroutines; i++ {
+		wg.Go(func() {
+			if ec.TryAdd(eventIDForTest) {
+				mu.Lock()
+				wins++
+				mu.Unlock()
+			}
+		})
+	}
+	wg.Wait()
+
+	// Assert
+	if wins != 1 {
+		t.Fatalf("expected exactly 1 TryAdd winner, got %d", wins)
+	}
+}
+
+// TestTryAddDebounceAndCooldown verifies that
+// 1- The first caller for a given eventID wins and sets the TTL
+// 2- concurrent callers during that TTL are rejected (debounced)
+// 3- after TTL expires, TryAdd succeeds again (cooldown expired)
+func TestTryAddDebounceAndCooldown(t *testing.T) {
+	ttl := 100 * time.Millisecond
+	ec, cleanupFunc, err := service.MakeTTLEventCache(ttl)
+	if err != nil {
+		t.Fatalf("MakeTTLCache error: %v", err)
+	}
+	defer func() { _ = cleanupFunc() }()
+
+	// first caller wins and sets the TTL
+	if !ec.TryAdd(eventIDInProgressForTest) {
+		t.Fatalf("expected first TryAdd to succeed")
+	}
+
+	done := make(chan bool, 1)
+	go func() {
+		done <- ec.TryAdd(eventIDInProgressForTest)
+	}()
+	addResult := <-done
+	if addResult {
+		t.Fatalf("expected concurrent TryAdd to be rejected while cooldown is active")
+	}
+
+	// after TTL elapses, TryAdd should succeed again
+	time.Sleep(ttl + 20*time.Millisecond)
```

Comment:
Don't use `time.Sleep` in tests if you are not using [`synctest`](https://pkg.go.dev/testing/synctest)